src/Transport.ts

```diff
@@ -543,26 +544,29 @@ export class Transport extends EnhancedEventEmitter
 
 		// This may throw.
 		const rtpParameters = ortc.getConsumerRtpParameters(
-			producer.consumableRtpParameters, rtpCapabilities!);
+			producer.consumableRtpParameters, rtpCapabilities!, pipe);
 
 		// Set MID.
-		rtpParameters.mid = `${this._nextMidForConsumers++}`;
-
-		// We use up to 8 bytes for MID (string).
-		if (this._nextMidForConsumers === 100000000)
+		if (!pipe)
 		{
-			logger.error(
-				`consume() | reaching max MID value "${this._nextMidForConsumers}"`);
+			rtpParameters.mid = `${this._nextMidForConsumers++}`;
+
+			// We use up to 8 bytes for MID (string).
+			if (this._nextMidForConsumers === 100000000)
+			{
+				logger.error(
+					`consume() | reaching max MID value "${this._nextMidForConsumers}"`);
 
-			this._nextMidForConsumers = 0;
+				this._nextMidForConsumers = 0;
+			}
 		}
 
 		const internal = { ...this._internal, consumerId: uuidv4(), producerId };
 		const reqData =
 		{
 			kind                   : producer.kind,
 			rtpParameters,
-			type                   : producer.type,
+			type                   : !pipe ? producer.type : 'pipe',
```

Comment:
cosmetic, just for readability:

`type: pipe ? pipe : producer.type`

Reply:
done